pkg/server/test/write_authzmodel.go

```diff
@@ -135,6 +136,21 @@ func WriteAuthorizationModelTest(t *testing.T, datastore storage.OpenFGADatastor
 				},
 			},
 		},
+		{
+			name: "self_referencing_type_restriction_with_entrypoint",
```

Comment:
why do we need this test here? isn't the test in `typesystem` enough? 🤔

Reply:
If the behavior of the WriteAuthorizationModel implementation changes relative to the typesystem (e.g. if we remove the typesystem usage), then WriteAuthorizationModel should still behave the same.